aten/src/ATen/functorch/BatchRulesScatterOps.cpp

```diff
@@ -977,11 +991,40 @@ std::tuple<Tensor,optional<int64_t>> index_add_batch_rule(
       other.select(*other_bdim, i) : other;
     const auto& index_slice = index_bdim.has_value() ?
       index.select(*index_bdim, i) : index;
-    results.push_back(at::index_add(self_slice, dim, index_slice, other_slice, alpha));
+
+    if (inplace) {
```

Comment:
Is there a simple way to not initialize `results` for inplace case (as it is not used then).

Or maybe we should guard call to reserve with `if (!inplace) { results.reserver(batch_size)}`

Wdyt?

Reply:
Done! I've guarded the initialization as you suggested.